common/sensor/sensor.c

```diff
@@ -57,189 +72,267 @@ bool access_check(uint8_t sensor_num)
+		status = isl69260_read(sensor_num, reading);
 		break;
-	case type_pch:
-		status = pal_pch_read(sensor_num, reading);
-		if (status)
-			return true;
+	case TYPE_PCH:
+		status = pch_read(sensor_num, reading);
 		break;
-	case type_hsc:
-		status = pal_hsc_read(sensor_num, reading);
-		if (status)
-			return true;
+	case TYPE_HSC:
+		status = adm1278_read(sensor_num, reading);
 		break;
-	case type_nvme:
-		status = pal_nvme_read(sensor_num, reading);
-		if (status)
-			return true;
+	case TYPE_NVME:
+		status = nvme_read(sensor_num, reading);
 		break;
-	case type_medusa:
+	case TYPE_MEDUSA:
 		//TODO: get sensors data of medusa
 		reading = 0;
-		status = 1;
-		if (status)
-			return true;
+		status = true;
 		break;
-	case type_fan:
+	case TYPE_FAN:
 		//TODO: get sensors data of fan
 		reading = 0;
-		status = 1;
-		if (status)
-			return true;
+		status = true;
 		break;
 
 	default:
 		printf("sensor_read with unexpected sensor type\n");
 		return false;
 		break;
 	}
-	return false;
+
+	if (cfg->post_sensor_read_fn) {
+		if (cfg->post_sensor_read_fn(sensor_num, cfg->post_sensor_read_args)) {
+			cfg->cache_status = SENSOR_FAIL_TO_ACCESS;
+			return false;
+		}
+	}
+
+	if (!status) {
+		cfg->cache_status = SENSOR_FAIL_TO_ACCESS;
+		return false;
+	}
+
+	return true;
 }
 
-uint8_t get_sensor_reading(uint8_t sensor_num, int *reading, uint8_t read_mode)
+uint8_t get_sensor_reading(uint8_t sensor_num, float *reading, uint8_t read_mode)
 {
+	if (reading == NULL) {
+		return false;
+	}
+
 	uint8_t status;
 
-	if (SnrNum_SDR_map[sensor_num] == 0xFF) { // look for sensor in SDR table
-		return SNR_NOT_FOUND;
+	if (sdr_index_map[sensor_num] == 0xFF) { // look for sensor in SDR table
+		return SENSOR_NOT_FOUND;
 	}
 
 	if (!access_check(sensor_num)) { // sensor not accessable
-		return SNR_NOT_ACCESSIBLE;
+		return SENSOR_NOT_ACCESSIBLE;
 	}
 
-	if (read_mode == get_from_sensor) {
+	if (read_mode == GET_FROM_SENSOR) {
 		status = sensor_read(sensor_num, reading);
 		if (status) {
 			if (!access_check(
 				    sensor_num)) { // double check access to avoid not accessible read at same moment status change
-				return SNR_NOT_ACCESSIBLE;
+				return SENSOR_NOT_ACCESSIBLE;
 			}
-			return sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache_status;
+			return sensor_config[sensor_config_index_map[sensor_num]].cache_status;
 		} else {
 			printf("sensor[%x] read fail\n", sensor_num);
-			return sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache_status;
+			return sensor_config[sensor_config_index_map[sensor_num]].cache_status;
 		}
-	} else if (read_mode == get_from_cache) {
-		if (sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache_status == SNR_READ_SUCCESS ||
-		    sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache_status ==
-			    SNR_READ_ACUR_SUCCESS) {
-			*reading = sensor_config[SnrNum_SnrCfg_map[sensor_num]].cache;
+	} else if (read_mode == GET_FROM_CACHE) {
+		if (sensor_config[sensor_config_index_map[sensor_num]].cache_status ==
```

Comment:
Please change this to a switch/case statement

Reply:
We have fixed this and done the forced update.
Please review it. Thanks.